Library/Homebrew/dev-cmd/bump-formula-pr.rb

```diff
@@ -331,7 +331,12 @@ def bump_formula_pr
 
     ohai "brew update-python-resources #{formula.name}"
     if !args.dry_run? || (args.dry_run? && args.write?)
-      PyPI.update_python_resources! formula, new_formula_version, silent: true, ignore_non_pypi_packages: true
+      begin
+        PyPI.update_python_resources! formula, new_formula_version, silent: true, ignore_non_pypi_packages: true
+      rescue Exception # rubocop:disable Lint/RescueException
+        formula.path.atomic_write(old_contents)
```

Comment:
I think this should still happen, though, if `--write` is passed.

In fact, none of the other `formula.path.atomic_write(old_contents) unless args.dry_run?`s have support for `--write`.

I'd change:

```patch
+ old_contents = File.read(formula.path) if !args.dry_run? || (args.dry_run? && args.write?)
- old_contents = File.read(formula.path) unless args.dry_run?
```

and


```patch
+ formula.path.atomic_write(old_contents) if !args.dry_run? || (args.dry_run? && args.write?)
- formula.path.atomic_write(old_contents) unless args.dry_run?
```

Truth table:

| `dry_run?` | `write?` | reset file on failure?
|---|---|---|
|✅|✅|✅|
|✅|❌|❌|
|❌|✅|✅ (n/a)|
|❌|❌|✅|

Reply:
What about:

```rb
read_only_run = args.dry_run && !args.write?
```

```rb
old_contents = File.read(formula.path) unless read_only_run
```

```rb
formula.path.atomic_write(old_contents) unless read_only_run
```

This would be useful for not repeating ourselves for the other instances of `formula.path.atomic_write(old_contents)`

Reply:
Yeah, I like that much better. I tried to use it before but some other maintainers found the `unless args.dry_run? && !args.write?` more to be more confusing than the longer version using `if`.

I think if it's extracted out into the `read_only_read` that makes it much more clear and so this an acceptable option (`do_something unless read_only_run` is probably easier to understand than `do_something unless
args.dry_run && !args.write?`)

Reply:
If you're making this change, you can also make it on this line: https://github.com/Homebrew/brew/blob/ce57d0bb33d2d35dd54a902ca70aab4edeea4221/Library/Homebrew/dev-cmd/bump-formula-pr.rb#L333

```ruby
    ohai "brew update-python-resources #{formula.name}"
    unless read_only_run
      begin
        ...
      rescue
        ...
      end
    end
```

Reply:
Note: `unless args.dry_run?` doesn't need to be added like in the Suggested change since it is already checked in the if statement above

Reply:
Oops, you're right!